pkg/controllers/management/aks/aks_cluster_handler.go

```diff
@@ -0,0 +1,696 @@
+	localCluster        = "local"
+	enqueueTime         = time.Second * 5
+)
+
+type aksOperatorValues struct {
+	HTTPProxy  string `json:"httpProxy,omitempty"`
+	HTTPSProxy string `json:"httpsProxy,omitempty"`
+	NoProxy    string `json:"noProxy,omitempty"`
+}
+
+type aksOperatorController struct {
+	clusterEnqueueAfter  func(name string, duration time.Duration)
+	secretsCache         wranglerv1.SecretCache
+	templateCache        v3.CatalogTemplateCache
+	projectCache         v3.ProjectCache
+	appLister            projectv3.AppLister
+	appClient            projectv3.AppInterface
+	nsClient             corev1.NamespaceInterface
+	clusterClient        v3.ClusterClient
+	clusterCache         v3.ClusterCache
+	catalogManager       manager.CatalogManager
+	systemAccountManager *systemaccount.Manager
+	dynamicClient        dynamic.NamespaceableResourceInterface
+	clientDialer         typesDialer.Factory
+}
+
+func Register(ctx context.Context, wContext *wrangler.Context, mgmtCtx *config.ManagementContext) {
+	aksClusterConfigResource := schema.GroupVersionResource{
+		Group:    aksAPIGroup,
+		Version:  "v1",
+		Resource: "aksclusterconfigs",
+	}
+
+	aksCCDynamicClient := mgmtCtx.DynamicClient.Resource(aksClusterConfigResource)
+	e := &aksOperatorController{
+		clusterEnqueueAfter:  wContext.Mgmt.Cluster().EnqueueAfter,
+		secretsCache:         wContext.Core.Secret().Cache(),
+		templateCache:        wContext.Mgmt.CatalogTemplate().Cache(),
+		projectCache:         wContext.Mgmt.Project().Cache(),
+		appLister:            mgmtCtx.Project.Apps("").Controller().Lister(),
+		appClient:            mgmtCtx.Project.Apps(""),
+		nsClient:             mgmtCtx.Core.Namespaces(""),
+		clusterClient:        wContext.Mgmt.Cluster(),
+		clusterCache:         wContext.Mgmt.Cluster().Cache(),
+		catalogManager:       mgmtCtx.CatalogManager,
+		systemAccountManager: systemaccount.NewManager(mgmtCtx),
+		dynamicClient:        aksCCDynamicClient,
+		clientDialer:         mgmtCtx.Dialer,
+	}
+
+	wContext.Mgmt.Cluster().OnChange(ctx, "aks-operator-controller", e.onClusterChange)
+}
+
+func (e *aksOperatorController) onClusterChange(key string, cluster *mgmtv3.Cluster) (*mgmtv3.Cluster, error) {
+	if cluster == nil || cluster.DeletionTimestamp != nil || cluster.Spec.AKSConfig == nil {
+		return cluster, nil
+	}
+
+	/*
+		if err := e.deployAKSOperator(); err != nil {
+			failedToDeployAKSOperatorErr := "failed to deploy aks-operator: %v"
+			var conditionErr error
+			if cluster.Spec.AKSConfig.Imported {
+				cluster, conditionErr = e.setFalse(cluster, apimgmtv3.ClusterConditionPending, fmt.Sprintf(failedToDeployAKSOperatorErr, err))
+				if conditionErr != nil {
+					return cluster, conditionErr
+				}
+			} else {
+				cluster, conditionErr = e.setFalse(cluster, apimgmtv3.ClusterConditionProvisioned, fmt.Sprintf(failedToDeployAKSOperatorErr, err))
+				if conditionErr != nil {
+					return cluster, conditionErr
+				}
+			}
+			return cluster, err
+		}
+
+	*/
```

Comment:
Uncomment this and reuse `clusteroperator.DeployOperator`

Reply:
Un-commented